index.html

```diff
@@ -123,97 +117,75 @@
-  <li>Set the <a>primary buffer</a> to a size of 150, unless otherwise specified by the user agent or set by the <a>setServerTimingBufferSize</a> method.</li>
+  <li>Let <dfn>entryList</dfn> be a new empty <a>Sequence</a>.
+  </li>
   <li>For each server-specified metric received via the <a>Server-Timing header field</a> perform the following steps:
-    <ol data-link-for="ServerEntry">
-      <li id="step-create-object">Create a new <a>PerformanceServerTiming</a> object and set <a>entryType</a> to the DOMString `server`.</li>
-      <li>Set the <a>name</a> to the <a data-cite="!HTML5#resolve-a-url">resolved URL</a> of the requested resource.</li>
+    <ol data-link-for="PerformanceServerTiming">
+      <li id="step-create-object">Let <dfn>entry</dfn> be a new <a>PerformanceServerTiming</a>.</li>
+      <li>Set <a>entryType</a> to the DOMString "`server`".</li>
+      <li>Set <a>name</a> to the <a data-cite="!HTML5#resolve-a-url">resolved URL</a> of the requested resource.</li>
       <li>Set <a>metric</a> to the server-specified metric name.</li>
-      <li>Set <a>duration</a> to the server-specified metric value, or value 0.</li>
+      <li>Set <a>duration</a> to the server-specified metric value, or value 0 if omitted or not representable as a double.</li>
       <li>Set <a>description</a> to the server-specified metric description, or an empty string.</li>
       <li>Set <a>startTime</a> to the value 0.</li>
-      <li>If the <a>primary buffer</a> is full, discard the <a>PerformanceServerTiming</a> object, created in step 3. Otherwise, add it to the <a>primary buffer</a>  . If adding it causes the <a> primary buffer</a> to become full, fire the <a>servertimingbufferfull</a>   event at the Document.
-        <ol>
-          <li>If the <a>clearServerTimings</a> method is called in the event handler for the <a>servertimingbufferfull</a> event, clear all <a>PerformanceServerTiming</a> objects in the <a>primary buffer</a>.</li  >
-          <li>If the <a>setServerTimingBufferSize</a> method is called in the event handler for the <a>servertimingbufferfull</a> event,
-          set the maximum size of the <a>primary buffer</a> to the `maxSize` parameter. If the `maxSize` parameter is less than the number of elements currently   stored in the buffer, no elements in the buffer are to be removed.</li>
-        </ol>
+      <li>Append <var>entry</var> to <var>entryList</var>.
       </li>
     </ol>
   </li>
+  <li>On the <a>PerformanceResourceTiming</a> object <a data-cite="!resource-timing#step-create-object">created for the
+    resource</a>, set <a>serverTiming</a> to <var>entryList</var>.
+  </li>
 </ol>
 
-The user-agent MUST process <a>Server-Timing header field</a> communicated via a trailer field (see [[!RFC7230]] section 4.1.2) using the same algorithm.</p>
+<p>The user-agent MUST process <a>Server-Timing header field</a> communicated via a trailer field (see [[!RFC7230]] section 4.1.2) using the same algorithm.</p>
```

Comment:
Maybe add a note or open an issue on that front. No need to block on getting consensus.